RAGchain/preprocess/text_splitter/token_splitter.py

```diff
@@ -0,0 +1,102 @@
+import copy
+from typing import List
+from uuid import uuid4
+
+from langchain.schema import Document
+from langchain.text_splitter import (CharacterTextSplitter, TokenTextSplitter, SpacyTextSplitter,
+                                     SentenceTransformersTokenTextSplitter, NLTKTextSplitter)
+from transformers import AutoTokenizer
+
+from RAGchain.preprocess.text_splitter.base import BaseTextSplitter
+from RAGchain.schema import Passage
+from RAGchain.utils.util import text_modifier
+
+
+class TokenSplitter(BaseTextSplitter):
+    """
+    Split a document into passages by recursively splitting on a list of separators.
+    You can specify a window_size and overlap_size to split the document into overlapping passages.
+    """
+
+    def __init__(self, tokenizer_name: str = 'tiktoken', chunk_size: int = 100, chunk_overlap: int = 0,
+                 pretrained_model_name: str = "gpt2", **kwargs):
+        """
+        :param tokenizer_name: A tokenizer_name name. You can choose tokenizer_name.
+                        (tiktoken, spaCy, SentenceTransformers, NLTK, huggingFace)
+        :param chunk_size: Maximum size of chunks to return. Default is 0.
+        :param chunk_overlap: Overlap in characters between chunks. Default is 0.
+        :param pretrained_model_name: A huggingface tokenizer pretrained_model_name to use huggingface token splitter.
+                                      You can choose various pretrained_model_name in this parameter. Default is "gpt2".
+                                      Refer to pretrained model in this link.  (https://huggingface.co/models)
+        :param kwargs: Additional arguments.
+        All splitters were inherited TextSplitter class in langchain text_splitter.py.
+        """
+        self.chosen_tokenizer = tokenizer_name
+
+        # tiktoken
+        self.tiktoken_splitter = TokenTextSplitter.from_tiktoken_encoder(chunk_size = chunk_size, chunk_overlap =  chunk_overlap)
+
+        # spaCy
+        self.spaCy_splitter = SpacyTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+
+        # SentenceTransformers (Default: chunk_overlap=0)
+        self.sentence_transformer_splitter = SentenceTransformersTokenTextSplitter(chunk_size=chunk_size,
+                                                                                   chunk_overlap=chunk_overlap)
+
+        # NLTK
+        self.NLTK_splitter = NLTKTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+        # Hugging Face
+        tokenizers = AutoTokenizer.from_pretrained(pretrained_model_name)
+        self.huggingFace_splitter = CharacterTextSplitter.from_huggingface_tokenizer(
+            tokenizers, chunk_size=chunk_size, chunk_overlap=chunk_overlap
+        )
```

Comment:
Make one splitter instance that user want to use. 
Plus, in this code spacy and nltk package's installation is not optional.